sagemaker-clarify/fairness_and_explainability/fairness_and_explainability_aws_sdk.ipynb

```diff
@@ -0,0 +1,889 @@
+    "                    \"OutputName\": \"analysis_result\",\n",
+    "                    \"S3Output\": {\n",
+    "                        \"S3Uri\": analysis_result_path,\n",
+    "                        \"LocalPath\": \"/opt/ml/processing/output\",\n",
+    "                        \"S3UploadMode\": \"EndOfJob\",\n",
+    "                    },\n",
+    "                }\n",
+    "            ]\n",
+    "        },\n",
+    "        ProcessingResources={\n",
+    "            \"ClusterConfig\": {\n",
+    "                \"InstanceCount\": 1,\n",
+    "                \"InstanceType\": \"ml.m5.xlarge\",\n",
+    "                \"VolumeSizeInGB\": 30,\n",
+    "            }\n",
+    "        },\n",
+    "        StoppingCondition={\n",
+    "            \"MaxRuntimeInSeconds\": 3600,\n",
+    "        },\n",
+    "        RoleArn=role,\n",
+    "    )\n",
+    "\n",
+    "    return processing_job_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here is a brief explanation of inputs used above, for detailed documentation check [CreateProcessingJob API reference](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateProcessingJob.html):\n",
+    "* AppSpecification: Here we provide the region specific clarify image uri we fethced earlier\n",
+    "* ProcessingInputs: Clarify job requires that you provde two ProcessingInput parameters.\n",
+    "  * The analysis configuration JSON file for a SageMaker Clarify job must be specified as an Amazon S3 object with the InputName \"analysis_config\". We will be providing the example analysis_configs that we have provided with this notebook. \n",
+    "  * InputName dataset which we are providing here as an Amazon S3 object.\n",
+    "* ProcessingOutputConfig: The job also requires an output parameter, the output location as an Amazon S3 prefix with the OutputName \"analysis_result\". The S3UploadMode should be set to \"EndOfJob\", because the analysis results is generated at the end of the job. We will be providing here the `analysis_result_path` that we configured earlier.\n",
+    "* ProcessingResources contains the ClusterConfig specifying the ML compute instance_type we want to use and the count. SageMaker SHAP analysis is CPU-intensive, using a better instance type should speed up the analysis. The SageMaker Clarify job doesn’t use GPU.\n",
+    "* StoppingCondition: Using a maximum limit of 30 min for example job run. You can set the MaxRuntimeInSeconds of a SageMaker Clarify job to up to 7 days (604800 seconds). If the job cannot be completed within this time limit, it will be force-stopped and no analysis results are provided."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "processing_job_name = create_processing_job(bias_analysis_config_path)\n",
+    "\n",
+    "# Wait for processing job to complete\n",
```

Comment:
Good to put the code snippet to a function so that it can be reused to wait for the explainability job as well.

And usually the job will run for 10 to 15 minutes, it is annoying to see many lines of InProgress status print. Good to use "." instead.

```
def wait_for_job(job_name):
    while sagemaker_client.describe_processing_job(ProcessingJobName=job_name)["ProcessingJobStatus"] == "InProgress":
        print(".", end="")
    print()
```

Then 

```
wait_for_job(processing_job_name)
```

should shows

```
..............
```

until the job completes.